src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -689,6 +689,16 @@ class BlockMatrix(val blocks: RDD[((Int, Int), BDM[Double])],
 
     new Table(hc, entriesRDD, rvRowType)
   }
+
+  def sparsify(keep: Array[Int]): M = {
```

Comment:
subsetPartitions returns an RDD with no partitioner. The block matrix constructor then fails to make a block matrix, because a partitioner is required. This seemed shorter than trying to add a partitioner to the RDD, but I can go back to that route. Why is this not considered a valid BlockMatrix?

Reply:
Because a BlockMatrix is assumed to have one block in each partition. Nearly all methods on BlockMatrix will throw assertion errors or worse if this is not the case.

Let’s discuss options after today’s meeting.